### PR TITLE
Let the user specify a starting vector for eigsh.

### PR DESCRIPTION
**Pull request by jmbr**
Sometimes a good initial guess for an eigenvector is known and it is advantageous to use it within the eigensolver. This patch adds the keyword argument ``v0`` to ``eigsh`` so that it behaves in the same way as in ``scipy.sparse.linalg``.

### CODE REVIEW

File: cupyx/scipy/sparse/linalg/_eigen.py [hunk 1]
```diff
@@ -79,8 +81,11 @@ def eigsh(a, k=6, *, which='LM', ncv=None, maxiter=None, tol=0,
     V = cupy.empty((ncv, n), dtype=a.dtype)
 
     # Set initial vector
-    u = cupy.random.random((n,)).astype(a.dtype)
-    V[0] = u / cublas.nrm2(u)
+    if v0 is None:
+        u = cupy.random.random((n,)).astype(a.dtype)
+        V[0] = u / cublas.nrm2(u)
+    else:
+        V[0] = cupy.asarray(v0)
```
Comment (kmaehashi) [inline cupyx/scipy/sparse/linalg/_eigen.py:88]: In CuPy, ndarray arguments must always be `cupy.ndarray` to avoid implicit host-to-device copy. So could you remove `cupy.asarray` here?

**Review by kmaehashi — CHANGES_REQUESTED** [review]
Thanks, LGTM except minor comment. Could you add test code for `v0 != None` case?

**Comment by jmbr** [comment]
> Thanks, LGTM except minor comment. Could you add test code for `v0 != None` case?

I have added the requested test in the last commit.

**Review by kmaehashi — APPROVED** [review]
Hi, sorry for the delay in review. LGTM!

**Comment by jmbr** [comment]
Hi, is there a pending step on my end to get this pull request merged? Please let me know if that's the case so I can help.

**Comment by kmaehashi** [comment]
Hi @jmbr, sorry for the delay! Test failures are irrelevant, so let's proceed to merge.

**Comment by YanivDorGalron** [comment]
Hey, 
Thanks for the helpful discussion
In the eigsh function i can determine the number of eigenvectors to calculate. 
how can i use v0 if i have a different initial guess for each of the k eigenvectors?
